scripts/w3cmanifestToLpf-script.js

```diff
@@ -0,0 +1,279 @@
+#! node
+
+/** Media type for JSON */
+const json_content_type = 'application/json';
+/** Media type for JSON-LD */
+const jsonld_content_type = 'application/ld+json';
+/** Media type for HTML */
+const html_content_type = 'text/html';
+
+const fetch = require('node-fetch');
+const { log, error } = require("console");
+const jsdom = require("jsdom");
+const { JSDOM } = jsdom;
+const { resolve } = require("url");
+const yazl = require("yazl");
+const { inflate } = require("zlib");
+const { publicEncrypt } = require("crypto");
+const fs = require("fs");
+const { promises } = require("dns");
+
+const progName = process.argv[1] || "";
+const argUrl = process.argv[2] || "";
+const packageName = process.argv[3] || "package.lpf";
+
+(async () => {
+
+
+    let publicationData;
+    let entryData;
+    let downloadStream;
+
+    try {
+        log(argUrl);
+        new URL(argUrl);
+
+    } catch (e) {
+        error(progName, "<URL>");
+        error(e.toString());
+
+        return;
+    }
+
+    const masterUrl = argUrl;
+
+    log("URL:", masterUrl);
+
+    log("fetching...");
+
+    let isJson = false;
+    try {
+
+        const response = await fetch(masterUrl);
+
+        log(response.ok);
+        log(response.status);
+        log(response.statusText);
+
+        const contentType = response.headers.get('content-type');
+        log(contentType);
+
+        const contentTypeArray = contentType.replace(/\s/g, '').split(';');
+        if (contentTypeArray.find((v) => v === json_content_type || v === jsonld_content_type)) {
+            isJson = true;
+
+            publicationData = await response.text();
+        } else if (contentTypeArray.includes(html_content_type)) {
+
+            isJson = false;
+            entryData = await response.text();
+        } else {
+            error("bad content-type");
+
+            return ;
+        }
+
+
+    } catch (e) {
+
+        error("error to fetch", masterUrl);
+        error(e);
+    }
+
+    log(publicationData || entryData);
+
+    // html parsing
+    if (!isJson) {
+        log("parse html file");
+
+        try {
+            const { window } = new JSDOM(entryData);
+
+            const elem = window.document.querySelector("link[rel=\"publication\"]");
+            if (elem) {
+                const href = elem.href;
+                log(href);
+
+                log("download the pubication linked in the html");
+
+
+                const publicationUrl = resolve(masterUrl, href);
+                log("publicationUrl", publicationUrl);
+
+                if (!publicationUrl.startsWith("about:blank")) {
```

Comment:
I don't see where you treat embedded manifests, identified by a fragment id in @href and manifest content in a script tag.

Reply:
you're right, I don't treat embedded manifest